Assembler.Base/BaseFrameHandler.cs

```diff
@@ -0,0 +1,38 @@
+﻿using Assembler.Core;
+using Assembler.Core.Entities;
+using Assembler.Core.Enums;
+
+namespace Assembler.Base
+{
+    public abstract class BaseFrameHandler<TFrame, TMessageInAssembly> : IFrameHandler<TFrame>
+        where TFrame : BaseFrame
+        where TMessageInAssembly : BaseMessageInAssembly
+    {
+        protected readonly ITimeBasedCache<TMessageInAssembly> TimeBasedCache;
+        protected readonly IFactory<TFrame, string> IdentifierFactory;
+        protected readonly IMessageEnricher<TFrame, TMessageInAssembly> MessageInAssemblyEnricher;
+        protected readonly ICreator<TMessageInAssembly> MessageInAssemblyCreator;
+        protected readonly IMessageReleaser<TMessageInAssembly> MessageReleaser;
+
+        protected BaseFrameHandler(ITimeBasedCache<TMessageInAssembly> timeBasedCache,
+            IFactory<TFrame, string> identifierFactory,
+            IMessageEnricher<TFrame, TMessageInAssembly> messageInAssemblyEnricher,
+            ICreator<TMessageInAssembly> messageInAssemblyCreator, IMessageReleaser<TMessageInAssembly> messageReleaser)
+        {
+            TimeBasedCache = timeBasedCache;
+            IdentifierFactory = identifierFactory;
+            MessageInAssemblyEnricher = messageInAssemblyEnricher;
+            MessageInAssemblyCreator = messageInAssemblyCreator;
+            MessageReleaser = messageReleaser;
+        }
+
+        public abstract void Handle(TFrame frame);
+
+        protected void ReleaseMessage(TMessageInAssembly message, ReleaseReason releaseReason) =>
+            MessageReleaser.Release(message, releaseReason);
+
+        protected string GetIdentifier(TFrame frame) => IdentifierFactory.Create(frame);
```

Comment:
I can't comment on the original discussion anymore so I'll write it here:
I though of something more like this:
```
private bool TryGetIdentifier(TFrame frame, ILogger logger, out string identifier)
{
    try
    {
        identifier = IdentifierFactory.Create(frame);
    }
    catch (Exception e)
    {
        logger.Error($"There was an error while getting an identifier out of the frame [{frame.Guid}], " +
                     $"it won't be used it the assembling process \n {e}");

        identifier = null;
        return false;
    }

    return true;
}
```
in the base.

Then you'll do: `if (!TryGetIdentifier(frame, logger, out string identifier)) return;` in the implementations

Reply:
I don't want to log in the Base class, we had some problems with it when I used it in the Rabbit